src/main/java/us/lsi/dp1/newcorporder/match/TakeOver.java

```diff
@@ -0,0 +1,75 @@
+package us.lsi.dp1.newcorporder.match;
+
+import com.google.common.base.Preconditions;
+import us.lsi.dp1.newcorporder.match.company.CompanyTile;
+import us.lsi.dp1.newcorporder.payload.request.CompanyAbilityRequest;
+import us.lsi.dp1.newcorporder.payload.request.ConsultantRequest;
+import us.lsi.dp1.newcorporder.payload.request.TakeOverRequest;
+
+public class TakeOver implements Move {
+    private final TurnSystem turnSystem;
+    private final Match match;
+
+    private ConsultantRequest consultantRequest;
+    private TakeOverRequest takeOverRequest;
+
+    public TakeOver(Match match) {
+        this.turnSystem = match.getTurnSystem();
+        this.match = match;
+    }
+
+    @Override
+    public void useConsultant(ConsultantRequest request) {
+        Preconditions.checkState(turnSystem.getCurrentState() == MatchTurnState.SELECTING_CONSULTANT, "must be in the current state!");
+        consultantRequest = request;
+        chooseConsultantToActivate(request.getConsultant());
+        turnSystem.setState(MatchTurnState.TAKING_OVER);
+    }
+
+    public void takeOver(TakeOverRequest request) {
+        Preconditions.checkState(turnSystem.getCurrentState() == MatchTurnState.TAKING_OVER, "must be in the current state!");
```

Comment:
```suggestion
        Preconditions.checkState(turnSystem.getCurrentState() == MatchTurnState.TAKING_OVER, "must be in the current state!");
        takeOverRequest = request;
```

Se os ha olvidado asignarlo y os va a dar `NullPointerException`.